data/modules/MissionUtils/OutfitRules.lua

```diff
@@ -0,0 +1,131 @@
+-- Copyright © 2008-2024 Pioneer Developers. See AUTHORS.txt for details
+-- Licensed under the terms of the GPL v3. See licenses/GPL-3.txt
+
+---@class MissionUtils.OutfitRules
+local OutfitRules = {}
+
+---@class MissionUtils.OutfitRule
+---@field slot string? Slot type filter string
+---@field equip string? Explicit equipment item to install in filtered slots
+---@field filter string? Filter for random equipment types
+---@field limit integer? Maximum number of equipment items to install
+---@field pick nil | "random" Pick the biggest/best item for the slot, or a random compatible item
+---@field maxSize integer? Limit the maximum size of items equipped by this rule
+---@field minSize integer? Limit the minimum size of items equipped by this rule
+---@field minThreat number? Minimum threat value for the entire ship that has to be met to consider this rule
+---@field maxThreatFactor number? Maximum proportion of remaining threat that can be consumed by this rule
+---@field randomChance number? Random chance to apply this rule, in [0..1]
+---@field balance boolean? Attempt to balance volume / threat across all slots this rule matches (works best with .pick = nil)
+
+OutfitRules.DifficultWeapon = {
+	slot = "weapon",
+	minSize = 2,
+	minThreat = 50.0,
+	maxThreatFactor = 0.7,
+	balance = true,
+}
+
+OutfitRules.ModerateWeapon = {
+	slot = "weapon",
+	limit = 2,
+	maxSize = 3,
+	minThreat = 30.0,
+	maxThreatFactor = 0.6,
+	balance = true,
+}
+
+OutfitRules.EasyWeapon = {
+	slot = "weapon",
+	maxSize = 2,
+	maxThreatFactor = 0.5,
+	balance = true,
+}
+
+OutfitRules.PulsecannonModerateWeapon = {
+	slot = "weapon",
+	filter = "weapon.energy.pulsecannon",
+	limit = 2,
+	maxSize = 3,
+	minThreat = 30.0,
+	maxThreatFactor = 0.6,
+	balance = true
+}
+
+OutfitRules.PulsecannonEasyWeapon = {
+	slot = "weapon",
+	filter = "weapon.energy.pulsecannon",
+	limit = 2,
+	maxSize = 3,
+	maxThreatFactor = 0.5,
+	balance = true
+}
+
+OutfitRules.DifficultShieldGen = {
+	slot = "shield",
+	minSize = 2,
+	minThreat = 50.0,
+	balance = true
+}
+
+OutfitRules.ModerateShieldGen = {
+	slot = "shield",
+	maxSize = 3,
+	limit = 2,
+	minThreat = 20.0,
+	maxThreatFactor = 0.8,
+	balance = true
+}
+
+OutfitRules.EasyShieldGen = {
+	slot = "shield",
+	maxSize = 2,
+	limit = 1,
+	minThreat = 20.0,
+	maxThreatFactor = 0.6
+}
+
+-- Default rules always equip the item if there's enough space
```

Comment:
Q? 

Not 100% sure how this works in conjunction with the `ShipTemplates.lua` which appear to add some of this equipment optionally.  Especially ShieldBooster and LaserCooler has me concerned as they can significantly increase the threat level..

Reply:
Note the pattern `utils.mixin(DefaultXXXRule, { minThreatLevel = X.X })` used in a lot of templates referencing the ShieldBooster and LaserCooler rules - this creates a copy of the default rule which only equips the item if there is enough baseline threat score.

For the most part, I've tried to replicate existing mission module logic almost exactly - a pass on mission difficulty would be definitely appreciated but is well out-of-scope for me at this time.

Reply:
Yes, I saw this later on. The comment is misleading..

(FWIW, there doesn't seem to be a "Close" button for a comment in this review; not sure how I can mark a discussion as resolved.. sorry)